backend/onyx/document_index/vespa/index.py

```diff
@@ -307,47 +314,61 @@ def register_multitenant_indices(
     def index(
         self,
         chunks: list[DocMetadataAwareIndexChunk],
-        fresh_index: bool = False,
     ) -> set[DocumentInsertionRecord]:
-        """Receive a list of chunks from a batch of documents and index the chunks into Vespa along
-        with updating the associated permissions. Assumes that a document will not be split into
-        multiple chunk batches calling this function multiple times, otherwise only the last set of
-        chunks will be kept"""
+        """
+        Index a list of chunks into Vespa, optionally removing older chunk versions.
+        Raises an AssertionError if multiple chunks for the same doc_id have conflicting doc_updated_at values.
+        """
+        current_timestamp = int(datetime.utcnow().timestamp())
+
+        # Step 1: Enforce a single doc_updated_at for each doc_id across all chunks.
+        doc_to_doc_updated_at = {}
```

Comment:
This is an interesting use of doc_updated_at ... I'm unclear on if it was being used for anything important before this that might be affected?  What are the tradeoffs vs using a new field?